hal/inc/cellular_hal_constants.h

```diff
@@ -0,0 +1,126 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+#ifndef CELLULAR_HAL_CONSTANTS_H
+#define CELLULAR_HAL_CONSTANTS_H
+
+#include "inet_hal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct __attribute__((__packed__))  _CellularConfig_t {
+    uint16_t size;
+    NetworkConfig nw;
+} CellularConfig;
+
+typedef int cellular_result_t;
+
+typedef int (*_CALLBACKPTR_MDM)(int type, const char* buf, int len, void* param);
+
+typedef void (*_CELLULAR_SMS_CB_MDM)(void* data, int index);
+
+typedef enum { CELLULAR_NETPROV_TELEFONICA =   0,
+               CELLULAR_NETPROV_TWILIO     =   1,
+               CELLULAR_NETPROV_UNKNOWN    = 999 } CellularNetProv;
```

Comment:
I ended up changing this to a CELLULAR_NETPROV_MAX enum, to be used only in the future if we need to cycle through providers.  Right now we test and set one of two.